docs/design/autotuing-design.md

```diff
@@ -0,0 +1,269 @@
+# Design to Auto-Tuning Hyperparameters with DLRover
+
+This design doc describes how DLrover auto-tunes hyperparameters a job.
+
+## Background
+
+In the fast-paced world of deep learning, large-scale language models have
+emerged as stars in the field of natural language processing. However, training
+these heavyweight models comes with a major challenge: finding the perfect
+training settings, especially when it comes to micro-batch sizes per GPU and
+learning rates. It's all about striking the right balance between speedy
+training and accurate results. With models getting bigger and optimization
+techniques getting trickier, predicting how much GPU memory need during
+training has become  tough and uncertain for user. That's why users often end
+up trying things out repeatedly to get the right settings.
+
+## Target
+
+Our primary objective is to design and implement a performance monitoring and
+optimization module for the training process, aimed at enhancing the efficiency
+ and effectiveness of distributed deep learning training. The specific goals
+encompass:
+
+ **Collecting Training Data:** We intend to monitor and gather
```

Comment:
Collecting Training Data ->  Collecting Runtime Metrics.

Training data is usually training datasets.

Reply:
Tks.I will pay more attention to the expressions. Fixed by latest commit.